framework/src/main/java/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -1910,18 +1903,14 @@ public void addComputedTypeAnnotations(Element elt, AnnotatedTypeMirror type) {
         applyQualifierParameterDefaults(elt, type);
         typeAnnotator.visit(type, null);
         defaults.annotate(elt, type);
-        if (dependentTypesHelper != null) {
-            dependentTypesHelper.standardizeVariable(type, elt);
-        }
+        dependentTypesHelper.standardizeVariable(type, elt);
     }
 
     @Override
     public ParameterizedExecutableType methodFromUse(MethodInvocationTree tree) {
         ParameterizedExecutableType mType = super.methodFromUse(tree);
         AnnotatedExecutableType method = mType.executableType;
-        if (dependentTypesHelper != null) {
-            dependentTypesHelper.viewpointAdaptMethod(tree, method);
-        }
+        dependentTypesHelper.viewpointAdaptMethod(tree, method);
```

Comment:
`#viewpointAdaptExecutable` doesn't call `#hasDependentAnnotations`  So now it is called even when the checker does not have any dependent type annotations.   Should `#viewpointAdaptExecutable` call  `#hasDependentAnnotations` ?  By making `dependentTypesHelper` non null for all checkers, there are several methods like this that are now called for every checker. 

How did you choose which methods have this check?  It seems to be most public methods, but not all.  Should it be all public methods?

Reply:
It's all the methods where there is currently a check; I moved the checks but did not introduce new ones.  The main reason was to avoid behavior change.  The minor reason was to avoid unnecessary checks.  I can add the checks to all the public methods.

Reply:
The previous design was that `dependentTypesHelper` is never used if the checker has no dependent type annotations.  The new design seems to be that methods in the `dependentTypesHelper` are called even if the checker has no dependent type, but the method immediately returns.  So it would make sense if all public methods do this check.  Otherwise, it would be easy to add a new call to a method that doesn't make this check.